kernel/kernel-api/src/main/java/io/delta/kernel/internal/TransactionBuilderImpl.java

```diff
@@ -150,16 +152,15 @@ public Transaction build(Engine engine) {
         metadata = metadata.withNewConfiguration(newProperties);
       }
 
-      Set<String> newWriterFeatures =
-          TableFeatures.extractAutomaticallyEnabledWriterFeatures(metadata, protocol);
+      Set<TableFeature> newWriterFeatures =
```

Comment:
Should it the variable be renamed to `newFeatures` since it's not writer-specific anymore?

Reply:
+1